flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/NettyMessageClientDecoderDelegateTest.java

```diff
@@ -0,0 +1,324 @@
+		for (ByteBuf buffer : inputBuffers) {
+			channel.writeInbound(buffer);
+		}
+
+		channel.runPendingTasks();
+
+		List<NettyMessage> decodedMessages = new ArrayList<>();
+		Object input;
+		while ((input = channel.readInbound()) != null) {
+			assertTrue(input instanceof NettyMessage);
+			decodedMessages.add((NettyMessage) input);
+		}
+
+		return decodedMessages;
+	}
+
+	private void verifyDecodedMessages(List<NettyMessage> expected, List<NettyMessage> decodedMessages) {
+		assertEquals(expected.size(), decodedMessages.size());
+		for (int i = 0; i < expected.size(); ++i) {
+			assertEquals(expected.get(i).getClass(), decodedMessages.get(i).getClass());
+
+			if (expected.get(i) instanceof NettyMessage.BufferResponse) {
+				BufferResponse expectedBufferResponse = (BufferResponse) expected.get(i);
+				BufferResponse decodedBufferResponse = (BufferResponse) decodedMessages.get(i);
+
+				assertEquals(expectedBufferResponse.backlog, decodedBufferResponse.backlog);
+				assertEquals(expectedBufferResponse.sequenceNumber, decodedBufferResponse.sequenceNumber);
+				assertEquals(expectedBufferResponse.bufferSize, decodedBufferResponse.bufferSize);
+				assertEquals(expectedBufferResponse.receiverId, decodedBufferResponse.receiverId);
+
+				if (expectedBufferResponse.bufferSize == 0 ||
+					expectedBufferResponse.receiverId.equals(RELEASED_CHANNEL_ID) ||
+					expectedBufferResponse.receiverId.equals(REMOVED_CHANNEL_ID)) {
+
+					assertNull(decodedBufferResponse.getBuffer());
+				} else if (expectedBufferResponse.receiverId.equals(NORMAL_CHANNEL_ID)) {
+					assertEquals(expectedBufferResponse.getBuffer(), decodedBufferResponse.getBuffer());
+				} else {
+					fail("The decoded buffer response was sent to unknown input channel.");
+				}
+			} else if (expected.get(i) instanceof NettyMessage.ErrorResponse) {
+				ErrorResponse expectedErrorResponse = (ErrorResponse) expected.get(i);
+				ErrorResponse decodedErrorResponse = (ErrorResponse) decodedMessages.get(i);
+
+				assertEquals(expectedErrorResponse.receiverId, decodedErrorResponse.receiverId);
+				assertEquals(expectedErrorResponse.cause.getClass(), decodedErrorResponse.cause.getClass());
+				assertEquals(expectedErrorResponse.cause.getMessage(), decodedErrorResponse.cause.getMessage());
+			} else {
+				fail("Unsupported message type");
+			}
+		}
+	}
+
+	private void releaseBuffers(ByteBuf[] buffers) {
+		for (ByteBuf buffer : buffers) {
+			buffer.release();
+		}
+	}
+
+	private Buffer createDataBuffer(int size) {
+		MemorySegment segment = MemorySegmentFactory.allocateUnpooledSegment(size);
+		NetworkBuffer buffer = new NetworkBuffer(segment, FreeingBufferRecycler.INSTANCE);
+		for (int i = 0; i < size / 4; ++i) {
+			buffer.writeInt(i);
+		}
+
+		return buffer;
+	}
+
+	private EmbeddedChannel createPartitionRequestClientHandler(int numberOfBuffersInNormalChannel) throws IOException, InterruptedException {
```

Comment:
createPartitionRequestClientHandler -> createChannelHandler and remove `InterruptedException`, then we can make this line short enough

Reply:
Another tip is placing this method right after `testRepartitionMessagesAndDecode`, then it is convenient for reviewing the codes avoid up and down repeately.